fusionbackend/src/main/java/io/fusion/fusionbackend/service/AssetSeriesService.java

```diff
@@ -71,46 +70,44 @@ public AssetSeries getAssetSeriesByCompany(final Long companyId, final Long asse
                 .orElseThrow(ResourceNotFoundException::new);
     }
 
-    public AssetSeries createAssetSeriesFromAssetTypeTemplate(final Long targetCompanyId,
-                                                              final Long assetTypeTemplateId) {
+    @Transactional
+    public AssetSeries createAssetSeries(final Long targetCompanyId, final Long assetTypeTemplateId,
+                                         final AssetSeries assetSeries) {
+
         final AssetTypeTemplate assetTypeTemplate =
                 assetTypeTemplateService.getAssetTypeTemplate(assetTypeTemplateId, true);
 
         final Company targetCompany = companyService.getCompany(targetCompanyId, false);
 
-        final AssetSeries newAssetSeries = AssetSeries.builder()
-                .build();
-        newAssetSeries.copyFrom(assetTypeTemplate);
-
-        assetTypeTemplate.getAssetSeries().add(newAssetSeries);
-        newAssetSeries.setAssetTypeTemplate(assetTypeTemplate);
-
-        newAssetSeries.setCompany(targetCompany);
-        targetCompany.getAssetSeries().add(newAssetSeries);
-        final AssetSeries savedAssetSeries = assetSeriesRepository.save(newAssetSeries);
-
-        List<FieldSource> newFieldSources = assetTypeTemplate.getFieldTargets().stream()
-                .map(fieldTarget ->
-                        FieldSource.builder()
-                                .fieldTarget(fieldTarget)
-                                .assetSeries(savedAssetSeries)
-                                .sourceUnit(fieldTarget.getField().getUnit())
-                                .build())
-                .collect(Collectors.toList());
-        List<FieldSource> savedFieldSources =
-                StreamSupport.stream(fieldSourceRepository.saveAll(newFieldSources).spliterator(), false)
-                        .collect(Collectors.toList());
-        newAssetSeries.getFieldSources().addAll(savedFieldSources);
-
-        return newAssetSeries;
+        assetSeries.setCompany(targetCompany);
```

Comment:
Shouldn't we add the new assetSeries to the list in company (bidirectional)? 
targetCompany.getAssetsSeries().add(assetSeries);